dulwich/index.py

```diff
@@ -516,7 +516,12 @@ def blob_from_path_and_stat(fs_path, st):
         with open(fs_path, 'rb') as f:
             blob.data = f.read()
     else:
-        blob.data = os.readlink(fs_path)
+        if sys.platform == 'win32':
+            fs_encoding = sys.getfilesystemencoding()
+            blob.data = (os.readlink(fs_path.decode(fs_encoding))
+                         .encode(fs_encoding))
```

Comment:
Yes :-( . This is the same for other types of files, because methods such as `open` will implicitly decode the path when passed bytes, using `sys.getfilesystemencoding()`, which will be `'mbsc'`, as the encoding. This is the cause of #203. 

This patch  will make symlinks behave the same as other files, which means the will work for ascii only, or mbsc encoded paths, but  will break for any other encoding (as per #203)

I make sure this is fixed when I get round to fixing #203.